api/src/main/java/net/kyori/adventure/text/KeybindComponent.java

```diff
@@ -57,6 +58,33 @@
   @Contract(pure = true)
   @NotNull KeybindComponent keybind(final @NotNull String keybind);
 
+  /**
+   * Sets the keybind.
+   *
+   * @param keybind the keybind
+   * @return a copy of this component
+   * @since 4.9.0
+   */
+  @Contract(pure = true)
+  default @NotNull KeybindComponent keybind(final @NotNull Keybind keybind) {
+    return this.keybind(Objects.requireNonNull(keybind, "keybind").keybind());
+  }
+
+  /**
+   * An object with a keybind.
+   *
+   * @since 4.9.0
+   */
+  interface Keybind {
```

Comment:
`KeybindSource`? Not sure about the naming.

Reply:
Yeah keybind source probably is better, the thing doesn't *have* to be a keybind after all.